### ISSUE TITLE
Question on the loss function formulation

### ISSUE THREAD

**Issue by hawkeyedesi** [mil-tokyo/dg_mmld#2]
Thanks for making your code public. I had a question about the loss function mentioned in the paper versus what has been implemented in the code: 

In the paper you have eq. 
![Screen Shot 2020-03-06 at 4 05 41 PM](https://user-images.githubusercontent.com/5407341/76131701-60c56780-5fc4-11ea-8dfe-1ec4cc7b20e1.png)

However, in the implementation I find the following: 
https://github.com/mil-tokyo/dg_mmld/blob/bfd5942cfc9bed3154e53ce5742955d75a119ce0/train/general.py#L44

I am trying to understand how to correlate the two. I would appreciate it if you could provide more details on how the implementation matches the equation in the paper.

**Comment by Toshihiko-Matsuura** [comment]
I am sorry for confusing you.
According to the implementation, the total training objective is rewritten as follows:
![image](https://user-images.githubusercontent.com/22876486/76141045-0dc2e180-60a4-11ea-9ede-d28882672ead.png)

In the implementation, the coefficient 'alpha' is set in the following line.
https://github.com/mil-tokyo/dg_mmld/blob/bfd5942cfc9bed3154e53ce5742955d75a119ce0/train/general.py#L20

In fact, the discriminator has the Gradient Reversal Layer (GRL), which inverse gradients by multiplying them with '- alpha'.
By doing this, eq. 4 is achieved by just one backpropagation.

**Comment by hawkeyedesi** [comment]
Thank you for the clarification. I can now correlate. 
Essentially you made the `Lambda` in eq 4 customizable for both the L_ent and the L_adv term by specifying the beta and alpha terms separately.

**Comment by fido20160817** [comment]
where is the implementation of style loss defined in equation(5)?
